### DISCUSSION TITLE
How to disable email confirmations?

### DISCUSSION THREAD

**Discussion by bliepp** [pglombardo/PasswordPusher#2046] [Q&A] [answered]
Hey there,

is it possible to deactivate the sending of email confirmations? I want to add a dummy account for external users (e.g. guests@guests.com), because I don't want everyone to be able to use my instance.

Best,
bliepp

**Comment by pglombardo** [comment]
Hi @bliepp - Just to be sure: you are running your own self-hosted instance and want to disable user account confirmations entirely?  Is that correct?

**Reply by bliepp** [reply]
Thanks for replying so quickly. Yeah, exactly. I'm running a self-hosted instance mainly for internal use and occasional communication with customers. As our server doesn't have much resources I wanted to restrict who can push passwords. To still provide the possibility for our customers to share passwords and API keys with us I was looking for some way to put the site for pushing passwords behind a password protection. My first idea was to create a fake account with temporary credentials so our customers can immediately use it, but that doesn't work since the email address needs to be confirmed.

Edit: The more I think about this, the dumber my request seems. The solution I thought of (i.e. creating non-confirmed accounts) is pretty inconvenient and I guess I'm just gonna leave it and either refer to your instance, make use of basic auth with my reverse proxy or simply activate anonymous submissions to the public.

**Reply by pglombardo** [reply, marked as answer]
Not a dumb request at all.  I'm planning some new features for this exact use case.  But for now, there isn't an easy way to remove the account confirmation requirement.

You could manually confirm accounts from the application console.  This directly modifies data though so be careful and maybe backup the database first.

1. Get a shell into the Docker container (`docker exec -it <container id> bash`)
2. `cd /opt/PasswordPusher`
3. `./bin/pwpush console`
4. `user = User.where(email: 'myemail@domain.com').first`
5. `user.confirm`
6. `user.save`

or alternatively, add a date/time to the `confirmed_at` field for the user in the [admin console](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#admin-dashboard) (`/admin`).
